### ISSUE TITLE
Show Type Hierarchy doesn't show subtypes

### ISSUE THREAD

**Issue by StellaHuang95** [microsoft/pylance-release#5403]
## Code Snippet

```python
class Base:
    def method(self):
        pass

class Derived(Base):
    def method(self):
        return super().method()
    
class Derived2(Base):
    def method(self):
        return super().method()
    
class Derived3(Derived, Derived2):
    def method(self):
        return super().method()

c = Derived2()
c.method()
```

## Repro Steps

1. place cursor on `Derived2` and see type hierarchy, check subtypes and supertypes

## Expected behavior

Both subtypes and supertypes are shown correctly. Specifically, when I click on `c.method()`, I expect it to show `method` in `Derived3` as its subtype.

## Actual behavior

No subtypes are shown.

**Comment by debonte** [comment]
Isn't this behavior (showing sub vs. super types) dependent on this toggle button?

![image](https://github.com/microsoft/pylance-release/assets/12820898/0eeecb50-41fe-44e7-a4b8-6754aa6214bd)

![image](https://github.com/microsoft/pylance-release/assets/12820898/97a629ec-e2f4-4f68-913f-185017a10914)

**Comment by StellaHuang95** [comment]
@debonte yes, this is the `show type hierarchy` feature, not the `show call hierarchy`. Show type hierarchy is supposed to show both overriding method and overriden method, so in this case, I am expecting `method` in `Derived3` to be shown as `c.method`'s subtype.
![image](https://github.com/microsoft/pylance-release/assets/100439259/e57a93f0-d026-4ad6-bf07-d39f0367245d)

**Comment by debonte** [comment]
Is that how it's supposed to work? Given that `Derived2.method` isn't a type, I wouldn't expect it to show overriding methods as subtypes. Maybe the bug here is that we shouldn't allow "Show Type Hierarchy" on non-type symbols?

**Comment by StellaHuang95** [comment]
Good point, this is just my interpretation of this [pr](https://github.com/microsoft/pyrx/pull/4285). 

@heejaechang might know it better. Regardless, I was expecting show supertype and subtype to behave consistently.

**Comment by Andrej730** [comment]
> Isn't this behavior (showing sub vs. super types) dependent on this toggle button?
> 
> ![image](https://github.com/microsoft/pylance-release/assets/12820898/0eeecb50-41fe-44e7-a4b8-6754aa6214bd)
> 
> ![image](https://github.com/microsoft/pylance-release/assets/12820898/97a629ec-e2f4-4f68-913f-185017a10914)

How do I reproduce this? In my case it seem to never show subtypes using the example below, only supertypes.

https://github.com/user-attachments/assets/9c2e06f2-6d6b-4475-a3e7-267492854b12

**Comment by Andrej730** [comment]
Is this still an issue? It seems to work now - `c.method` subtypes is showing `Derived3.method`

https://github.com/user-attachments/assets/12c518ce-9e54-46e3-9888-225c2cded6fc

**Comment by debonte** [comment]
@heejaechang, looks like this was fixed by https://github.com/microsoft/pyrx/pull/5151?

**Comment by heejaechang** [comment]
no longer repo in latest pylance.
